### PR TITLE
feat: support specify post alerts timeout

### PR DESCRIPTION
**Pull request by zjj2wry**
I want to specify push alerts timeout. currently, Prometheus config supports it. so I create a pr for it. thanks

https://prometheus.io/docs/prometheus/latest/configuration/configuration/#alertmanager_config

### CODE REVIEW

**Comment by zjj2wry** [comment]
image

tested it in my env

File: pkg/apis/monitoring/v1/types.go [hunk 1]
```diff
@@ -657,6 +657,8 @@ type AlertmanagerEndpoints struct {
 	// Version of the Alertmanager API that Prometheus uses to send alerts. It
 	// can be "v1" or "v2".
 	APIVersion string `json:"apiVersion,omitempty"`
+	// Per-target Alertmanager timeout when pushing alerts.
```
Comment (lilic) [inline pkg/apis/monitoring/v1/types.go:660]: Nit: Can you instead do:
```suggestion
	// Timeout is a per-target Alertmanager timeout when pushing alerts.
```

Comment (lilic) [reply pkg/apis/monitoring/v1/types.go:660]: Is this a v1 Alertmanager field only?

Comment (zjj2wry) [reply pkg/apis/monitoring/v1/types.go:660]: > Is this a v1 Alertmanager field only?

it also supports in v2

**Comment by zjj2wry** [comment]
> Thanks for your PR! We are in the middle of moving the repo to a new org so tests might be failing a bit and you might need to rebase.

cool~I have been using kube-prometheus and prometheus-operator in a production environment for a while now, thanks for the community 💯

**Comment by zjj2wry** [comment]
> Thanks so much, this looks good to me, do you just mind squashing the commits into two, one for feature and one for generate. Thanks!
> 
> 🎉

squashed, thanks

**Comment by lilic** [comment]
Thanks for this contribution! 🎉

**Pull request merged**